custom_components/elasticsearch/es_doc_creator.py

```diff
@@ -115,7 +117,7 @@ def _state_to_attributes(self, state: State) -> dict:
 
             # so we replace them with an "_" instead.
             # https://github.com/legrego/homeassistant-elasticsearch/issues/92
-            key = str.replace(orig_key, ".", "_")
```

Comment:
Yeah this does increase the risk of a key collision since we're collapsing the set of allowed characters. Logging a warning like you suggest is probably a good enough for now. We can explore a more sophisticated approach if the need arises.

Reply:
Done in https://github.com/legrego/homeassistant-elasticsearch/pull/251/commits/37c70143c61d1f56bac9cacd55f112da3b66ccef